src/Billing/StatementsClient.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace ArrowSphere\PublicApiClient\Billing;
+
+use ArrowSphere\PublicApiClient\Billing\Entities\Statement;
+use ArrowSphere\PublicApiClient\Billing\Enum\ConstantEnum;
+use ArrowSphere\PublicApiClient\Exception\EntityValidationException;
+use ArrowSphere\PublicApiClient\Exception\NotFoundException;
+use ArrowSphere\PublicApiClient\Exception\PublicApiClientException;
+use Generator;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * Class StatementsClient
+ */
+class StatementsClient extends AbstractBillingClient
+{
+    /**
+     * @param string $reference
+     *
+     * @return Statement|null
+     *
+     * @throws EntityValidationException
+     * @throws NotFoundException
+     * @throws PublicApiClientException|\ReflectionException
+     */
+    public function getStatement(string $reference): ?Statement
+    {
+        $response = $this->getStatementRaw($reference);
+        $data = $this->decodeResponse($response);
+        $result = null;
+        if ($data['data'] && $data['data']['billingStatement']) {
+            $result = new Statement($data['data']['billingStatement']);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param string $reference
+     *
+     * @return string
+     *
+     * @throws NotFoundException
+     * @throws PublicApiClientException
+     */
+    public function getStatementRaw(string $reference): string
+    {
+        $reference = urlencode($reference);
+        $this->path = "/statements/$reference";
+
+        return $this->get();
+    }
+
+    /**
+     * Lists the statements.
+     * Returns an array (generator) of Statement.
+     *
+     * @param string $periodFrom YYYY-MM
+     * @param string $periodTo YYYY-MM
+     *
+     * @return Generator
```

Comment:
`@return Generator|Statement[]`
If psalm whines about it, make it ignore this error with `make psalm-generate-baseline`